client-spark/common/src/main/java/org/apache/spark/shuffle/RssShuffleHandle.java

```diff
@@ -69,19 +69,56 @@ public ShuffleDependency<K, V, C> getDependency() {
     return dependency;
   }
 
-  public Map<Integer, List<ShuffleServerInfo>> getPartitionToServers() {
-    return partitionToServers;
-  }
-
   public int getShuffleId() {
     return shuffleId();
   }
 
-  public Set<ShuffleServerInfo> getShuffleServersForData() {
-    return shuffleServersForData;
+  /**
+   * Tried to get cached {@link ShuffleHandleInfo} from local thread first and then memory if not existing in local.
+   * If not cached, one of competing threads gets chance to deserialize it and caches it for other threads.
+   *
+   * @return
+   */
+  private ShuffleHandleInfo getCurrentHandleInfo() {
+    // local first
+    ShuffleHandleInfo info = _localHandleInfo.get().get();
```

Comment:
Why do we need localHandleInfo  and currentHandleInfo? If we run two shuffle at the same time, it will not reduce the time of serialization.

Reply:
They are for caching deserialized HandleInfo for the current active shuffle.  All tasks pertaining to a shuffle must be completed before running another shuffle in next stage. So no two shuffles are at same time.

Reply:
We can run two shuffles at the same time, such as join operation.

Reply:
Spark can also run multiple jobs at the same time.

Reply:
You are correct. Thanks for pointing it out. Let me fix it.

Reply:
yes, I deserialize here because I use the same way as Spark broadcasts task which get serialized first by Java serializer into bytearray then by KyroSerializer in TorrentBroadcast.  

Spark already registered bytearry to KyroSerializer. 

I think it's not user friend if we ask they register RssShuffleHandle class.

Reply:
Yes, it's disabled by default. I forgot to mention another factor that the broadcast variable (ShuffleHandleInfo, not RssShuffleHandle) is about 4MB without JavaSerializer.